### PR TITLE
CPDP-2152: add support to switch between languages

### PR DESCRIPTION
**Pull request by mikail-khan**

### CODE REVIEW

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/components/language_switcher.scala.html [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        <a id="switch-language" href="@routes.CustomLanguageController.switchToEnglish" data-journey-click="link - click:Language selector:English">
+            <span class="visuallyhidden">@messages("language-switcher.english.hidden")</span>
+            <span aria-hidden="true">@messages("language-switcher.english")</span></a>
+        <span aria-hidden="true"> | </span>
+        <span>@messages("language-switcher.welsh")</span>
+        }
+    </li>
+</ul>
+</nav>
+}
```
Comment (mikail-khan) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/components/language_switcher.scala.html:47]: @nubz added this in just to test - change as required

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/govuk_wrapper.scala.html [hunk 2]
```diff
@@ -140,6 +141,9 @@
 
 @mainContentHeader = {
     @contentHeader.fold(Html(""))(header => uiMainContentHeader(contentHeader = header))
+    <div class="centered-content">
+        @languageSwitcher(appConfig.enableLanguageSwitching)
+    </div>
 }
```
Comment (mikail-khan) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/govuk_wrapper.scala.html:147]: @nubz added in for now just to test - needs to be relocated as appropriate.

File: conf/messages.cy [hunk 3]
```diff
@@ -0,0 +1,3935 @@
+#===================================================
+# TRANSLATION NOT REQUIRED = START
+#===================================================
+
```
Comment (mikail-khan) [inline conf/messages.cy:4]: @nubz @AndyHWChung copied the current messages in just for testing purposes - should I take this out?

Comment (AndyHWChung) [reply conf/messages.cy:4]: you could maybe take only the non-agent start page content and the onboarding check your details page content  and replace the english content with random crap. Then we can see clearly if we toggle the language on the start page whether or not it works.

Comment (mikail-khan) [reply conf/messages.cy:4]: I tested it locally and the crap works

Comment (mikail-khan) [reply conf/messages.cy:4]: I'm going to take all this out and switch it off

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/govuk_wrapper.scala.html [hunk 4]
```diff
@@ -140,6 +141,9 @@
 
 @mainContentHeader = {
     @contentHeader.fold(Html(""))(header => uiMainContentHeader(contentHeader = header))
+    <div class="centered-content">
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/govuk_wrapper.scala.html:144]: if the feature isn't enabled it'd be cleaner not to have an empty `div` here

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/CustomLanguageController.scala [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+) extends LanguageController(configuration, languageUtils, cc) {
+
+  private val english: Lang = Lang("en")
+  private val welsh: Lang   = Lang("cy")
+
+  override def fallbackURL: String = configuration.underlying.get[String]("self.url").value
+
+  override def languageMap: Map[String, Lang] = Map("en" -> english, "cy" -> welsh)
+
+  def switchToEnglish: Action[AnyContent] = switchToLanguage("en")
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/CustomLanguageController.scala:39]: does this redirect you back to the page you were originally on when it's done its thing?

Comment (mikail-khan) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/CustomLanguageController.scala:39]: yes

File: conf/messages.cy (hunk 3 above)
Comment (AndyHWChung) [reply conf/messages.cy:4]: for the purposes of QA'ing it, it'd be easier to test if you leave at least a couple of pages with some random translations - when we eventually start getting translations through we can then replace the crap with the actual thing

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/govuk_wrapper.scala.html (hunk 2 above)
Comment (nubz) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/govuk_wrapper.scala.html:145]: I removed the unnecessary wrapping div (which has undesired effect of nesting centred content with unwanted padding inside an already centred parent)

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/govuk_wrapper.scala.html (hunk 4 above)
Comment (nubz) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/govuk_wrapper.scala.html]: that's removed - not wanted anyway - not sure why HTS use it as it offsets the right alignment with unwanted padding

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/components/language_switcher.scala.html (hunk 1 above)
Comment (nubz) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/components/language_switcher.scala.html:47]: LGTM

File: conf/messages.cy (hunk 3 above)
Comment (mikail-khan) [reply conf/messages.cy]: just translated the account home page

**Comment by AndyHWChung** [comment]
LGTM

**Pull request merged**